webapp/channels/src/components/team_controller/team_controller.tsx

```diff
@@ -86,10 +86,15 @@ function TeamController(props: Props) {
         function handleFocus() {
             if (props.selectedThreadId) {
                 window.isActive = true;
+                props.markThreadAsRead(props.selectedThreadId);
```

Comment:
This component doesn't need the current thread/channel/post IDs except for marking things as read on focus. To avoid passing those into the `TeamController` and the re-rendering that causes, could we perhaps rename `markChannelAsReadOnFocus` to just `markAsReadOnFocus` and have that contain both its current logic plus both calls to `markThreadAsRead`? Doing that would also avoid adding and removing these listeners as often since we wouldn't do it on every channel/thread change.

Ideally, we'd do something similar for `fetchChannelsAndMembers` below as well, but that's got a ref which makes that a bit more difficult, so I don't think we need to bother with it now

Reply:
Also, for `window.isActive`, I think we can just set that to true in `handleFocus` without checking if there's a selected thread or current channel first.

It seems like a bug that we don't set it when the window is focused outside of one of those contexts. That if statement was added in https://github.com/mattermost/mattermost-webapp/pull/4839, but it seems like that was a mistake